components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/JdbcUtils.java

```diff
@@ -29,17 +29,39 @@
  */
 public class JdbcUtils {
 
+    /**
+     * Enum to select the database type.
+     */
+    public enum Database {
+        IDENTITY, SESSION
+    }
+
     /**
      * Get a new Jdbc Template.
      *
      * @return a new Jdbc Template.
      */
+    @Deprecated
     public static JdbcTemplate getNewTemplate() {
 
         DataSource dataSource = IdentityDatabaseUtil.getDataSource();
         return new JdbcTemplate(dataSource);
     }
 
+    /**
+     * Get a new Jdbc Template.
+     *
+     * @return a new Jdbc Template.
+     */
+    public static JdbcTemplate getNewTemplate(Database database) {
+
+        DataSource dataSource = IdentityDatabaseUtil.getDataSource();
```

Comment:
DataSource dataSource =null;
        if (Database.SESSION.equals(database)) {
            dataSource = IdentityDatabaseUtil.getSessionDataSource();
        } else { dataSource = IdentityDatabaseUtil.getDataSource();
      }
if(dataSource == null) {throw error,} else {
        return new JdbcTemplate(dataSource);}

The reason is that previous impl takes two datasources unnecessarily.